contracts/beacon/Voucher.sol

```diff
@@ -263,12 +269,18 @@ contract Voucher is Initializable, IVoucher {
     }
 
     /**
-     * Retrieve the expiration timestamp of the voucher.
-     * @return expirationTimestamp The expiration timestamp.
+     * Drain balance of voucher on PoCo if it is expired.
+     * Funds are sent to the VoucherHub contract.
+     * @param amount amount to be drained
      */
-    function getExpiration() external view returns (uint256) {
+    function drain(uint256 amount) external onlyVoucherHub onlyExpired {
         VoucherStorage storage $ = _getVoucherStorage();
-        return $._expiration;
+        address voucherHub = $._voucherHub;
+        // Although transfer function in PoCo always returns true (or reverts),
+        // a return value check is added here in case its behavior changes.
+        if (!IERC20(IVoucherHub(voucherHub).getIexecPoco()).transfer(voucherHub, amount)) {
```

Comment:
`transfer(voucherHub, amount)` -> `transfer(voucherHub, IERC20(iexecPoco).balanceOf(this))`

Reply:
Doing it this way consumes 500 more gas.
The `amount` information is already available in the VoucherHub which is the unique caller of this function that's why I chose to implement it that way.

Reply:
You are mixing things with the VCHR check

Reply:
What am I mixing exactly?
https://github.com/iExecBlockchainComputing/iexec-voucher-contracts/compare/feature/drain...feature/drain-without-amount?expand=1

Reply:
The link you just sent looks good, but I don't understand your message!
You said `amount` was available in hub, but it was VCHR amount, not SRLC amount :)

Reply:
They are both strictly equal.

Reply:
At everypoint in time, for the same voucher:
```
VCHR amount === SRLC amount
```

Reply:
No because of potential remainders during claims

Reply:
Ok, so you're talking about your new PR that is still under review?
1. I don't see the case where the two amounts would be different even with remainders. Remainder must be exactly the same in RLC and in VCHR, otherwise we're doing something wrong.
2. If it's the case, I think it's a bad idea to allow this situation to appear, we can even prohibit sponsoring when sponsored amount is not a multiple of the volume. 
3. I don't think we should block this PR because of the other one. The current implementation is based on the condition that amounts in VCHR and in RLC are the same. If this condition changes after your PR we should update the drain function accordingly, later.

Reply:
No I don't agree with you, SRLC & VCHR are not equal at any point, another example is anyone can send at any point SRLC to the voucher.

Reply:
Exactly, that's why VCHR should be our reference for operations such as the drain. We should not take more than what was funded.

Reply:
Yes that's something we need to think about.